sumologic/resource_sumologic_token.go

```diff
@@ -0,0 +1,117 @@
+package sumologic
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceSumologicToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSumologicTokenCreate,
+		Read:   resourceSumologicTokenRead,
+		Update: resourceSumologicTokenUpdate,
+		Delete: resourceSumologicTokenDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"version": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
can `type` have any other value apart from `CollectorRegistration`? If not, then we should add a check for the same.

Reply:
Currently, we only have `CollectorRegistration` type for tokens.
But the tricky part is that the type will change between request and response.
In the request, the type for token is `CollectorRegistration`. However, in the response, type will be `CollectorRegistrationTokenResponse`. This is rendered by content library framework.

For example, the request body is:
```
{
  "name": "test-token",
  "description": "new token",
  "status": "Active",
  "type": "CollectorRegistration",
  "version": 0
}
```
And response will be like this:
```
{
    "id": "0000000000008A31",
    "name": "test-yuting",
    "description": "new token",
    "status": "Active",
    "type": "CollectorRegistrationTokenResponse",
    "version": 0,
    "createdAt": "2020-07-08T17:21:30.732Z",
    "createdBy": "0000000005C48CF8",
    "modifiedAt": "2021-06-10T17:10:43.489Z",
    "modifiedBy": "000000000B05888D",
    "encodedTokenAndUrl": "xxx"
}
```

Reply:
Had some discussions offline on the validation for type.
Decided to add type check for `CollectorRegistration` and `CollectorRegistrationTokenResponse`.